cmake/modules/Buildc-ares.cmake

```diff
@@ -14,7 +14,7 @@ function(build_c_ares)
   add_library(c-ares::c-ares STATIC IMPORTED)
   add_dependencies(c-ares::c-ares c-ares_ext)
   set_target_properties(c-ares::c-ares PROPERTIES
-    INTERFACE_INCLUDE_DIRECTORIES "${C-ARES_SOURCE_DIR}"
+    INTERFACE_INCLUDE_DIRECTORIES "${C-ARES_SOURCE_DIR};${C-ARES_BINARY_DIR}"
```

Comment:
They are defined in `./cmake/modules/Buildc-ares.cmake` as:
```cmake
  set(C-ARES_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/c-ares")
  set(C-ARES_BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/c-ares")
```

In my understanding the value of `CMAKE_CURRENT_BINARY_DIR` is, relatively to the repo's top directory, `build/src` while `CMAKE_CURRENT_SOURCE_DIR` – `src`.

It looks the header really sits there:

```
$ find -name ares_build.h
./build/src/c-ares/ares_build.h
$ ls build/src/c-ares
ares_build.h   CMakeCache.txt       lib
ares_config.h  CMakeFiles           Makefile
bin            cmake_install.cmake  Makefile.inc.cmake
```

I was a little bit confused by `..._BINARY_DIR`. Initially, I thought it points to `build/src/c-ares/bin`.

Reply:
the "current" dir of a function is defined by its caller, which is sitting in "src". so `CMAKE_CURRENT_BINARY_DIR` is `$CMAKE_BINARY_DIR/src`.  with the "c-ares" component, "c-ares_ext" will be built in `$CMAKE_BINARY_DIR/src/c-ares`.

actually, you could refer `src/c-ares/CMakeLists.txt` to find an authoritative answer
```cmake
# Write ares_build.h configuration file.  This is an installed file.
CONFIGURE_FILE (ares_build.h.cmake ${PROJECT_BINARY_DIR}/ares_build.h)
...
    TARGET_INCLUDE_DIRECTORIES (${PROJECT_NAME} PUBLIC "$<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>")
    TARGET_INCLUDE_DIRECTORIES (${PROJECT_NAME} PUBLIC "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>")
```

Reply:
and thank you for the review!